src/main/java/dog/pawbook/model/managedentity/dog/DateOfBirth.java

```diff
@@ -0,0 +1,79 @@
+package dog.pawbook.model.managedentity.dog;
+
+import static dog.pawbook.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a Dog's breed in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDob(String)}
+ */
+public class DateOfBirth {
+
+    public static final String MESSAGE_CONSTRAINTS = "Date Of Birth should only "
+            + "contain numbers and should not be blank.";
+    public static final String VALIDATION_REGEX = "^(?:(?:31(\\/|-|\\.)(?:0?[13578]|1[02]))\\1|(?:(?:29|30)(\\/|-|\\.)"
+            + "(?:0?[1,3-9]|1[0-2])\\2))(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$|^(?:29(\\/|-|\\.)0?2\\3"
+            + "(?:(?:(?:1[6-9]|[2-9]\\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))"
+            + "$|^(?:0?[1-9]|1\\d|2[0-8])(\\/|-|\\.)(?:(?:0?[1-9])|(?:1[0-2]))\\4(?:(?:1[6-9]|[2-9]\\d)?\\d{2})$";
```

Comment:
is this going to match all kinds of date formats or just our specified one?

Reply:
This will take in dd/mm/yyyy, dd-mm-yyyy or dd.mm.yyyy and helps validate leap year as well. Do you want me to modify it to only support dd/mm/yyyy?

Reply:
Nope this is perfect. But just need to make sure that the LocalDate parser is able to handle this variety. Maybe add a comment above it to indicate what it is looking out for? It will help make it more readable.